### DISCUSSION TITLE
Events emails not sent

### DISCUSSION THREAD

**Discussion by refugeindhamma** [liedekef/events-made-easy#214] [Q&A] [answered]
All email settings have been checked, including the Wordpress admin email, and everything is in place.

Event emails are not being sent even though test emails are successful. There are no errors with my email system but nothing is being sent for single events and recurring events. EME shows entries in "Pending Bookings" after a registration is being made. "Sent Emails" under "Email Management" also shows that the emails have been sent but nothing is sent when I checked my email service provider sent log.

What else can I check for this?

**Comment by liedekef** [comment]
Depending on your mail config in EME: once EME calls the configured method for sending mail, it is no longer in EME hands.
In this case, it might be that the sender is not allowed by your mailserver, but you should not look in the sent logs, but your mailserver logs as a whole (since they are not sent, the mailserver probably dropped/bounced them and you should look there).
Check your configured sender in EME, best is to force the sender to be a known address in the same domain as your webserver.

**Reply by refugeindhamma** [reply]
Instead of "sent log" I should have said "Mailing History". There are no bounces.
![image](https://github.com/liedekef/events-made-easy/assets/101639308/13f5f220-ea2b-431b-94a6-b4c41b8e448c)

Test emails are sent successfully.
![Screenshot 2023-06-26 at 22-43-34 Email management](https://github.com/liedekef/events-made-easy/assets/101639308/352c39d2-2bda-4a7e-98e3-d65fda77426a)

Received successfully.
![Screenshot 2023-06-26 224627](https://github.com/liedekef/events-made-easy/assets/101639308/89e73520-1e3e-4991-8cd2-7773f3cf2675)

I also used the same email address used for test email in the registration but can't receive any emails. Tried other email addresses and all failed. Another person using Microsoft email also did not receive anything after doing a registration test.

Edit: Removed email addresses from images.

**Reply by liedekef** [reply]
What are your settings in EME for sender?
Also: you can activate SMTP debug (since it seems you're using SMTP) and do a registration then, it should show the complete trace for mail.

**Reply by refugeindhamma** [reply]
Where can I find the email log for EME after doing a registration? The previous one posted was shown in "Test Email".

**Reply by refugeindhamma** [reply]
PHP_FPM is giving a lot of EME errors.
![Screenshot 2023-06-26 235700](https://github.com/liedekef/events-made-easy/assets/101639308/0fea67dc-7cc7-4244-9feb-37467a3d7d31)
![Screenshot 2023-06-26 235545](https://github.com/liedekef/events-made-easy/assets/101639308/c43454fb-4697-4496-a5d3-d1e7b84e944f)

**Reply by liedekef** [reply]
Those warnings are situated in this code:

```
        $generic_mail_from_email = get_option( 'eme_mail_sender_address' );
        $generic_mail_from_name  = get_option( 'eme_mail_sender_name' );
        if ( ! eme_is_email( $generic_mail_from_email ) ) {
                $contact                 = eme_get_contact();
                $generic_mail_from_email = $contact->user_email;
                $generic_mail_from_name  = $contact->display_name;
        }
```

Meaning: your EME settings for generic email is empty (or not a valid email), so EME tries to get a default admin contact via eme_get_contact. That function looks at the EME setting for default contact person, it seems even that is not set correctly (or the user that was set in it has been deleted in the mean time). The next version will try to fix that, see this changeset: https://github.com/liedekef/events-made-easy/commit/e2f15cc14734b9808fc1d05ddcdecf9398cd1537

Now the other settings (changing SSL etc ...) are not relevant (if your mailserver supports them, make sure to use the correct port for SSL or TLS).

**Comment by refugeindhamma** [comment]
I need to change from "SMTP" at "Email sending method" to "Wordpress Email (default)" then it works. Then, if I switch it back to "SMTP", the event emails can get sent out. But if I change the "SMTP encryption method" to "TLS", it stops working again. Even If I switch it to "SSL", it will not work until I change to ""Wordpress Email (default)", either leaving it as it is or switch it back to "SMTP" > "SSL", then it will continue working.

Edit: The "SMTP encryption method" was initially set to "TLS" when the problem occurred.

Edit 2: The problem is not with my email provider's TLS. I changed Fluent SMTP's setting to "TLS" and it works on both Fluent SMTP and EME.

**Comment by refugeindhamma** [comment, marked as answer]
Franky, my understanding of EME SMTP correct?

Things to check when emails are not being sent out when using "SMTP" option under EME "Settings" > "Email" > "Email sending method". 

1) Double check "SMTP" settings under EME "Settings" > "Email" > "SMTP Host" to "SMTP Password".

2) If "Default email sender address" is empty or not forced and "SMTP" settings are correct,
a)  EME "Email Management" > "Test Email" will send out emails using "Administration Email Address". So, test emails will be successfully sent out but Event emails will fail to send out if "Contact Person" email address is not permitted/approved by Email Service Provider.
Check "Contact Person" email address to be Email Service Provider's permitted/approved "Sender Email Address".
This can be found in Wordpress "Users" > "All Users" > Name of Contact Person for EME Events.

b) If test emails fail to send, it is a problem with "Administration Email Address". 
Check Wordpress "Settings" > "General" > "Administration Email Address" to be Email Service Provider's permitted/approved "Sender Email Address".

3) If "Default email sender address" is filled in and/or forced and "SMTP" settings are correct,
a) EME "Email Management" > "Test Email" will send out emails using "Default email sender address". So, if emails fail to send, it is a problem with "Default email sender address".
Check EME "Settings" > "Email" > "Default email sender address" to be Email Service Provider's permitted/approved "Sender Email Address".

Edit:
Reply from Franky below:
> EME has no notion of Wordpress "Users" > "All Users" > Name of Contact Person for EME Events ... Test email and regular email for events work in the same way, the only difference is that for an event the email of the contact person is taken as sender (if the sender is not forced in the EME settings). The default contact person (for events) is taken for the test email as sender in that case too (since a test email is not linked to an event I need something to go on). This logic has nothing to do with SMTP, but your mail sending method (if wordpress mail or php mail is chosen) can set a fixed "from" if they so desire (this is not in scope of EME, I can't change fixed php configs then). SMTP allows more freedom but needs correct settings of course ...

So, to test Event email sending, I can use "Event Related Email" and EME will use the Contact Person as specified in the Event to send the emails.

**Comment by liedekef** [comment]
EME has no notion of  Wordpress "Users" > "All Users" > Name of Contact Person for EME Events ...
Test email and regular email for events work in the same way, the only difference is that for an event the email of the contact person is taken as sender (if the sender is not forced in the EME settings). The default contact person (for events) is taken for the test email as sender in that case too (since a test email is not linked to an event I need something to go on).
This logic has nothing to do with SMTP, but your mail sending method (if wordpress mail or php mail is chosen) can set a fixed "from" if they so desire (this is not in scope of EME, I can't change fixed php configs then). SMTP allows more freedom but needs correct settings of course ...

**Reply by refugeindhamma** [reply]
So if I want to test Event email sending, I can use "Event Related Email" and EME will use the Contact Person as specified in the Event?

**Reply by liedekef** [reply]
That's also a method.
